File: 45-jump-game-ii/45-jump-game-ii.cpp
class Solution {
public:
    int jump(vector<int>& nums) {
        
        int len = nums.size();
        vector<int> dp(len, 0); 
        // store the number of minimum jumps to reach ith index
        
        for (int i = 0; i < len; i++) {
            int curMax = min(len - 1, i + nums[i]);
            int jump = dp[i] + 1;
            
            for (int j = i + 1; j <= curMax; j++) {
                if (dp[j]) continue;
                dp[j] = jump;
                if(j == len - 1) break;
            }
        }
        
        return dp[len - 1];
    }
};

File: 45-jump-game-ii/45-jump-game-ii.py
class Solution:
    def jump(self, nums: List[int]) -> int:
        
        numsLen = len(nums)
        
        jmp = [987654321 for x in range(numsLen)]
        jmp[0] = 0
        
        for pos in range(0, numsLen):
            
            for i in range(pos + 1, min(pos + nums[pos] + 1, numsLen)):
                jmp[i] = min(jmp[i], jmp[pos] + 1)
                
        print(jmp)
                
        return jmp[numsLen - 1]
        